rpc/service.go

```diff
@@ -21,66 +21,65 @@ import (
 
 type WaitFunc func(context.Context)
 
-type CertConfig struct {
+type AuthConfig struct {
 	TLS         bool
+	UserAuth    bool
 	SkipVerify  bool
 	CAName      string
 	ServiceName string
 	Depot       certdepot.Depot
 	UserManager gimlet.UserManager
 }
 
-func (c *CertConfig) Validate() error {
-	if c.Depot == nil {
-		return errors.New("must specify a certificate depot")
-	}
-	if c.CAName == "" {
-		return errors.New("must specify a CA name")
-	}
-	if c.ServiceName == "" {
-		return errors.New("must specify a service name")
-	}
+func (c *AuthConfig) Validate() error {
+	catcher := grip.NewBasicCatcher()
+
+	catcher.AddWhen(c.TLS && c.UserAuth, errors.New("must specify either TLS or user auth, not both"))
```

Comment:
Should this also check for `!c.TLS && !c.UserAuth` so we don't end up in a state with no authentication?

Reply:
No auth is useful for testing.